tsconfig.build.json

```diff
@@ -4,6 +4,7 @@
         "./tsconfig-build-options.json"
     ],
     "compilerOptions": {
+        "allowImportingTsExtensions": true,
```

Comment:
I'm not sure if this is controversial @turadg. I enabled it because ESLint was complaining about unresolved imports in `patterns` following the TypeScript migration.

Reply:
I'm on my phone but without looking over the whole PR I wonder if the imports would work with .js instead. Usually TS can figure out the correct module. We'd like to avoid this option in this repo because it makes it easier to accidentally ship modules that have TS syntax.

Reply:
Based on the number of CI failures due to this, I think I will have to walk it back anyway.

Reply:
Ah, your comments didn't show up until a refresh. tl;dr: I walked it back to `.js`. Not sure why I got the ESLint errors in the first place but they're gone now.